### PR TITLE
agent: enable ssl only for kvm agent (not in system vms)

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes the issue that template cannot be downloaded from https site in 4.17.0.0 RC2.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [x] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

File: agent/src/main/java/com/cloud/agent/AgentShell.java [hunk 1]
```diff
@@ -399,12 +401,17 @@ public void init(String[] args) throws ConfigurationException {
...
         final File agentFile = PropertiesUtil.findConfigFile("agent.properties");
         if (agentFile == null) {
             s_logger.info("Failed to find agent.properties file");
             return;
         }
+        final String resource = getProperty(null, "resource");
+        if (!LIBVIRT_COMPUTING_RESOURCE.equalsIgnoreCase(resource)) {
+            s_logger.info("This is not a cloudstack kvm agent, ignoring");
+            return;
+        }
```
Comment (rohityadavcloud) [inline agent/src/main/java/com/cloud/agent/AgentShell.java:414]: @nvazquez @weizhouapache Could this cause an issue with (a) direct download, (b) ssl enabled cpvm and ssvm. Should we attempt to fix the issue by importing or referencing the system keystore too where java/ca certs are stored and imported (for x1/letsencrypt) @Pearl1594 ?

Comment (weizhouapache) [reply agent/src/main/java/com/cloud/agent/AgentShell.java:414]: @rohityadavcloud @nvazquez @Pearl1594 
this part of code was introduced in #6200 which aims to fix an issue with direct download on KVM.
with this pr, SSL is still enabled on KVM hosts, but not in system VMs where Java process load key store real hostile.keystone (same as the behavior without #6200 )

**Comment by weizhouapache** [comment]
@blueorangutan ubuntu20 kvm-ubuntu20

**Review by nvazquez — APPROVED** [review]

**Comment by nvazquez** [comment]
Verified HTTPS templates working on KVM env with this fix

File: agent/src/main/java/com/cloud/agent/AgentShell.java [hunk 2]
```diff
@@ -399,12 +401,17 @@ public void init(String[] args) throws ConfigurationException {
...
 
-    private void enableSSL() {
+    private void enableSSLForKvmAgent() {
         final File agentFile = PropertiesUtil.findConfigFile("agent.properties");
         if (agentFile == null) {
             s_logger.info("Failed to find agent.properties file");
             return;
         }
+        final String resource = getProperty(null, "resource");
+        if (!LIBVIRT_COMPUTING_RESOURCE.equalsIgnoreCase(resource)) {
```
Comment (rohityadavcloud) [inline agent/src/main/java/com/cloud/agent/AgentShell.java:411]: The use of `LIBVIRT_COMPUTING_RESOURCE` or `com.cloud.hypervisor.kvm.resource.LibvirtComputingResource` may be misleading (for ex. what if this isn't the resource name? for example, we do support lxc etc?).

Can you simply add a check if the agent is either is in systemvm (for ex. /etc/cloudstack-release would exist) or if we're on a KVM host (check for /dev/kvm exists or libvirt running)? (Depending on what we're trying to achieve here). The other option usually is to check the resource class via some method.

What do you think @weizhouapache @nvazquez ?

File: agent/src/main/java/com/cloud/agent/AgentShell.java (hunk 1 above)
Comment (rohityadavcloud) [reply agent/src/main/java/com/cloud/agent/AgentShell.java:414]: Makes sense if this was relatively newer code.

File: agent/src/main/java/com/cloud/agent/AgentShell.java (hunk 2 above)
Comment (weizhouapache) [reply agent/src/main/java/com/cloud/agent/AgentShell.java:411]: @rohityadavcloud 
I will move all these codes to `LibvirtComputingResource` which makes more sense.
I will verify the issue fixed by #6200 as well.

**Comment by weizhouapache** [comment]
@blueorangutan package

File: agent/src/main/java/com/cloud/agent/AgentShell.java (hunk 2 above)
Comment (nvazquez) [reply agent/src/main/java/com/cloud/agent/AgentShell.java]: +1

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by slavkap — APPROVED** [review]
tested - LGTM

**Review by Pearl1594 — APPROVED** [review]
LGTM

**Review by nvazquez — APPROVED** [review]
LGTM, manually tested

**Comment by nvazquez** [comment]
@blueorangutan test

**Pull request merged**